client/my-sites/importer/author-mapping-pane.jsx

```diff
@@ -30,12 +32,90 @@ export default React.createClass( {
 		targetTitle: PropTypes.string.isRequired
 	},
 
+	getUserCount() {
+		const { totalUsers } = UsersStore.getPaginationData( {
+			siteId: this.props.siteId,
+			order: 'ASC',
+			order_by: 'display_name',
+			number: 50
+		} );
+
+		return totalUsers;
+	},
+
 	render: function() {
 		const { hasSingleAuthor, sourceAuthors, sourceTitle, targetTitle, onMap, onStartImport, siteId } = this.props;
 		const canStartImport = hasSingleAuthor || sourceAuthors.some( author => author.mappedTo );
+		const targetUserCount = this.getUserCount();
+		let mappingDescription;
+
+		if ( targetUserCount === 1 && sourceAuthors.length === 1 ) {
+			mappingDescription = this.translate(
+				'We found one author on your %(sourceType)s site. ' +
+				'Because you\'re the only author on {{b}}%(destinationSiteTitle)s{{/b}}, ' +
+				'all imported content will be assigned to you. ' +
+				'Click Start Import to proceed.', {
+					args: {
+						sourceType: 'WordPress',
+						destinationSiteTitle: targetTitle
+					},
+					components: {
+						b: <strong />
+					}
+				}
+			);
+		} else if ( targetUserCount === 1 && sourceAuthors.length > 1 ) {
+			mappingDescription = this.translate(
+				'We found multiple authors on your %(sourceType)s site. ' +
+				'Because you\'re the only author on {{b}}%(destinationSiteTitle)s{{/b}}, ' +
+				'all imported content will be assigned to you. ' +
+				'Click Start Import to proceed.', {
+					args: {
+						sourceType: 'WordPress',
+						destinationSiteTitle: targetTitle
+					},
+					components: {
+						b: <strong />
+					}
+				}
+			);
+		} else if ( targetUserCount > 1 && sourceAuthors.length === 1 ) {
+			mappingDescription = this.translate(
+				'We found multiple authors on {{b}}%(destinationSiteTitle)s{{/b}}. ' +
+				'Please reassign the authors of the imported items to an existing ' +
+				'user on {{b}}%(destinationSiteTitle)s{{/b}}, then click Start Import.', {
+					args: {
+						sourceType: 'WordPress',
+						destinationSiteTitle: targetTitle
+					},
+					components: {
+						b: <strong />
+					}
+				}
+			);
+		} else if ( targetUserCount > 1 && sourceAuthors.length > 1 ) {
+			mappingDescription = this.translate(
+				'We found multiple authors on your %(sourceType)s site. ' +
+				'Please reassign the authors of the imported items to an existing ' +
+				'user on {{b}}%(destinationSiteTitle)s{{/b}}, then click Start Import.', {
+					args: {
+						sourceType: 'WordPress',
+						destinationSiteTitle: targetTitle
+					},
+					components: {
+						b: <strong />
+					}
+				}
+			);
+		}
 
 		return (
 			<div className="importer__mapping-pane">
+				<SiteUsersFetcher fetchOptions={ { siteId: this.props.siteId, number: 1, order: 'ASC', order_by: 'display_name' } }>
```

Comment:
> Intriguing - what does number: 1 imply here?
It will fetch only 1 user, which is all we need to access the pagination data. It's a little frustrating that you have to fetch at least one user to get access to `paginationData`. The API returns that, total_users, etc, with each fetch. Ideally, there should be a separate endpoint to get total user count. Maybe we could flesh that out, but in the interest of time, I figure work with what we've got.

> @ebinnion warned me about sending a different number count for different fetches as it could duplicate network calls.

`number` is omitted from `fetchOptions` when we decide if we need to call the server again, so I don't think a server call will be duplicated.

> Is there enough duplication here to warrant pulling this object out into a reusable object/function?

I think yes!